Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -630,5 +630,47 @@ TEST_F(SerializerTest, EncodesEmptyDocument) {
   ExpectRoundTrip(key, empty_value, update_time, proto);
 }
 
+TEST_F(SerializerTest, EncodesNonEmptyDocument) {
+  DocumentKey key = DocumentKey::FromPathString("path/to/the/doc");
+  FieldValue fields = FieldValue::ObjectValueFromMap({
+      {"foo", FieldValue::StringValue("bar")},
+      {"two", FieldValue::IntegerValue(2)},
+      {"nested", FieldValue::ObjectValueFromMap({
+                     {"fourty-two", FieldValue::IntegerValue(42)},
+                 })},
+  });
+  SnapshotVersion update_time = SnapshotVersion{{1234, 5678}};
+
+  google::firestore::v1beta1::Value inner_proto;
+  google::protobuf::Map<std::string, google::firestore::v1beta1::Value>*
```

Comment:
Entirely optional: using a reference instead of a pointer would eliminate the need to dereference the pointer below.